woocommerce/woocommerce
# Use API package to create orders by batch

### All Submissions:


### Changes proposed in this Pull Request:

Currently in `trunk`, the e2e test `order-status-filters.test.js` creates test orders through the UI. This PR replaces these setup steps to use the ["Batch Create Orders" API](https://woocommerce.github.io/woocommerce-rest-api-docs/#batch-update-orders) to improve the speed of this test.

Based on the latest CI runs in `trunk` and this branch, notice that we've halved the execution time of this test:
<img width="634" alt="Screen Shot 2021-07-01 at 1 19 31 PM" src="https://user-images.githubusercontent.com/4509348/124069839-725f3800-da6f-11eb-92d0-9ea9a29a0bc8.png">
<img width="634" alt="Screen Shot 2021-07-01 at 1 20 40 PM" src="https://user-images.githubusercontent.com/4509348/124069856-7a1edc80-da6f-11eb-954d-8a62ac177f8a.png">


Closes #29797.

### How to test the changes in this Pull Request:

1. Launch the e2e test environment
2. Run the `activate-and-setup` tests: `npx wc-e2e test:e2e ./tests/e2e/specs/activate-and-setup/setup-onboarding.js` 
3. Run the test that this PR changed.
    Headless mode: `npx wc-e2e test:e2e ./tests/e2e/specs/wp-admin/order-status-filters.test.js`
    UI  mode: `npx wc-e2e test:e2e-dev ./tests/e2e/specs/wp-admin/order-status-filters.test.js`
4. They should both pass
<img width="457" alt="Screen Shot 2021-06-30 at 7 01 26 PM" src="https://user-images.githubusercontent.com/4509348/123949825-9bcb8580-d9d5-11eb-9c90-bda0bf9ef941.png">

### Other information:




### Changelog entry

In `core-tests`:
> Updated the `order-status-filters.test.js` e2e test to use the API in creating orders

In `e2e-utils`:
> New function to create orders by batch using the orders API
